File: server/src/knowledge/ingestors/audioIngestor.js
import { BaseIngestor } from '../baseIngestor.js';

// Target design: local speech-to-text (whisper.cpp via Ollama or a dedicated
// binary) producing timestamped segments, chunked like a transcript.
export class AudioIngestor extends BaseIngestor {
  constructor() {
    super('audio');
  }

  async ingest(_source) {
    throw new Error('AudioIngestor not implemented — needs local whisper transcription');
  }
}

File: server/app/knowledge/ingestors/audio_ingestor.py
from ..base_ingestor import BaseIngestor


class AudioIngestor(BaseIngestor):
    def __init__(self):
        super().__init__("audio")

    async def ingest(self, source: dict) -> list[dict]:
        raise NotImplementedError("AudioIngestor not implemented — needs local whisper transcription")
